<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>640</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:1, stop:0 rgba(152, 252, 152, 255), stop:1 rgba(240, 246, 218, 255));
}

QPushButton {
    width: 75px;
    color: black;
 	background-color: #ddffff;
    border-top: 2px solid #eeffff;
    border-right: 2px solid #ffddff;
    border-bottom: 2px solid #ccffff;
    border-left: 2px solid #ffffdd;
    border-radius: 5px;
}

QPushButton:hover {
	color: black;
    background-color: aqua;
}

QPushButton:pressed {
    color: white;
 	background-color: #0000ff;
    border-top: 2px solid #ccffff;
    border-right: 2px solid #ffffdd;
    border-bottom: 2px solid #eeffff;
    border-left: 2px solid #ffddff;
}

QPushButton:disabled {
    color: gray;
 	background-color: gainsboro;
    border-top: 2px solid #f0f0f0;
    border-right: 2px solid lightgray;
    border-bottom: 2px solid silver;
    border-left: 2px solid whitesmoke;
}

QStatusBar {
 	background-color: gainsboro;
    border-top: 2px solid silver;
    border-right: 2px solid whitesmoke;
    border-bottom: 2px solid white;
    border-left: 2px solid rightgray;
}

QDialog {
    background-color: white;
}</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="FileList" name="fileListWidget">
        <property name="styleSheet">
         <string notr="true">QListWidget {
  background-color: lightcyan;
  border-radius: 5px;
  border-top: 4px solid cyan;
  border-right: 4px solid mintcream;
  border-bottom: 4px solid azure;
  border-left: 4px solid cyan;
}

QListWidget::item:selected {
    color: navy;
 	background-color: cyan;
}
</string>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SelectionMode::ContiguousSelection</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="aboutButton">
          <property name="text">
           <string>About</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="addButton">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Add</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="upButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Up</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="downButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Down</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="removeButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Remove</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="clearButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="createPushButton">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>Create PLS</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Folder:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="FileDirEdit" name="folderLineEdit">
        <property name="styleSheet">
         <string notr="true">QLineEdit {
 	background-color: gainsboro;
    border-top: 2px solid silver;
    border-right: 2px solid whitesmoke;
    border-bottom: 2px solid white;
    border-left: 2px solid rightgray;
}</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="folderButton">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>Folder</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>FileName: </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="FileDirEdit" name="fileNameLineEdit">
        <property name="styleSheet">
         <string notr="true">QLineEdit {
 	background-color: ghostwhite;
    border-top: 2px solid silver;
    border-right: 2px solid whitesmoke;
    border-bottom: 2px solid white;
    border-left: 2px solid rightgray;
}</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="closeButton">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="text">
         <string>Close</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>FileList</class>
   <extends>QListWidget</extends>
   <header>filelist.h</header>
   <slots>
    <signal>Message2StatusBar(QString)</signal>
    <signal>ButtonEnabled(bool)</signal>
    <signal>RemoveEnabled(bool)</signal>
    <signal>UpEnabled(bool)</signal>
    <signal>DownEnabled(bool)</signal>
    <slot>UpSelected()</slot>
    <slot>DownSelected()</slot>
    <slot>RemoveSelected()</slot>
    <slot>AddFiles()</slot>
    <slot>ItemsSelectChanged()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>FileDirEdit</class>
   <extends>QLineEdit</extends>
   <header location="global">filediredit.h</header>
   <slots>
    <signal>SetFile(QString)</signal>
    <signal>Message2StatusBar(QString)</signal>
    <slot>SetFolder()</slot>
    <slot>ForceSetText(QString)</slot>
    <slot>TextSpecified()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>clearButton</sender>
   <signal>clicked()</signal>
   <receiver>fileListWidget</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>788</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>522</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileListWidget</sender>
   <signal>Message2StatusBar(QString)</signal>
   <receiver>statusbar</receiver>
   <slot>showMessage(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>449</y>
    </hint>
    <hint type="destinationlabel">
     <x>212</x>
     <y>587</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileListWidget</sender>
   <signal>ButtonEnabled(bool)</signal>
   <receiver>createPushButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>650</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>745</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileListWidget</sender>
   <signal>ButtonEnabled(bool)</signal>
   <receiver>clearButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>679</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>788</x>
     <y>248</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileListWidget</sender>
   <signal>UpEnabled(bool)</signal>
   <receiver>upButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>631</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel">
     <x>788</x>
     <y>170</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileListWidget</sender>
   <signal>DownEnabled(bool)</signal>
   <receiver>downButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>670</x>
     <y>73</y>
    </hint>
    <hint type="destinationlabel">
     <x>727</x>
     <y>195</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileListWidget</sender>
   <signal>RemoveEnabled(bool)</signal>
   <receiver>removeButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>668</x>
     <y>111</y>
    </hint>
    <hint type="destinationlabel">
     <x>788</x>
     <y>222</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>upButton</sender>
   <signal>clicked()</signal>
   <receiver>fileListWidget</receiver>
   <slot>UpSelected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>734</x>
     <y>159</y>
    </hint>
    <hint type="destinationlabel">
     <x>655</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>downButton</sender>
   <signal>clicked()</signal>
   <receiver>fileListWidget</receiver>
   <slot>DownSelected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>788</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>94</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeButton</sender>
   <signal>clicked()</signal>
   <receiver>fileListWidget</receiver>
   <slot>RemoveSelected()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>721</x>
     <y>220</y>
    </hint>
    <hint type="destinationlabel">
     <x>453</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>closeButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>765</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>673</x>
     <y>557</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>folderButton</sender>
   <signal>clicked()</signal>
   <receiver>folderLineEdit</receiver>
   <slot>SetFolder()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>558</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>559</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>folderLineEdit</sender>
   <signal>SetFile(QString)</signal>
   <receiver>fileNameLineEdit</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>144</x>
     <y>552</y>
    </hint>
    <hint type="destinationlabel">
     <x>444</x>
     <y>559</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>folderLineEdit</sender>
   <signal>Message2StatusBar(QString)</signal>
   <receiver>statusbar</receiver>
   <slot>showMessage(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>560</y>
    </hint>
    <hint type="destinationlabel">
     <x>106</x>
     <y>595</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addButton</sender>
   <signal>clicked()</signal>
   <receiver>fileListWidget</receiver>
   <slot>AddFiles()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>750</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>594</x>
     <y>18</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileNameLineEdit</sender>
   <signal>editingFinished()</signal>
   <receiver>fileNameLineEdit</receiver>
   <slot>TextSpecified()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>492</x>
     <y>561</y>
    </hint>
    <hint type="destinationlabel">
     <x>517</x>
     <y>562</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>createPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>CreatePLSFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>788</x>
     <y>537</y>
    </hint>
    <hint type="destinationlabel">
     <x>763</x>
     <y>458</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileNameLineEdit</sender>
   <signal>Message2StatusBar(QString)</signal>
   <receiver>statusbar</receiver>
   <slot>showMessage(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>469</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>470</x>
     <y>590</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fileListWidget</sender>
   <signal>itemSelectionChanged()</signal>
   <receiver>fileListWidget</receiver>
   <slot>ItemsSelectChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>680</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel">
     <x>642</x>
     <y>286</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>aboutButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>ShowAboutDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>769</x>
     <y>16</y>
    </hint>
    <hint type="destinationlabel">
     <x>790</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>CreatePLSFile()</slot>
  <slot>ShowAboutDialog()</slot>
 </slots>
</ui>
